go/api/base_client.go

```diff
@@ -71,3 +87,43 @@ func (client *baseClient) Close() {
 	C.close_client(client.coreClient)
 	client.coreClient = nil
 }
+
+func (client *baseClient) executeCommand(requestType C.RequestType, args []string) (interface{}, error) {
+	if client.coreClient == nil {
+		return nil, &ClosingError{"Unable to execute requests; the client is closed. Please create a new client."}
+	}
+
+	cArgs := toCStrings(args)
+	defer freeCStrings(cArgs)
+
+	resultChannel := make(chan payload)
+	resultChannelPtr := uintptr(unsafe.Pointer(&resultChannel))
+
+	C.command(client.coreClient, C.uintptr_t(resultChannelPtr), requestType, C.uintptr_t(len(args)), &cArgs[0])
+
+	payload := <-resultChannel
+	if payload.error != nil {
+		return nil, payload.error
+	}
+
+	if payload.value != nil {
+		return *payload.value, nil
+	}
+
+	return nil, nil
```

Comment:
Shouldn't we always return one or the other?  Maybe this is a `panic`?

Reply:
Direction from Amazon has always been if there's an unexpected error in the payload, this is an extreme issue and we should close the client.

Reply:
This was supposed to cover the case where the redis command was successful but the return value was null. I realized I can rewrite this to be simpler. I'll do so now, which will remove this statement.